Code/Framework/AzFramework/AzFramework/Components/NativeUISystemComponent.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Components/NativeUISystemComponent.h>
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace AzFramework
+{
+    NativeUISystemComponent::NativeUISystemComponent()
+    {
+        if ((m_applicationImplFactory = CreateApplicationImplementationFactory()) != nullptr)
+        {
+            AZ::Interface<Application::ImplementationFactory>::Register(m_applicationImplFactory);
+        }
```

Comment:
The AZ::Interface template has been updated, so that it doesn't assert internally anymore when an error occurs when registering or unregistering in this [commit](https://github.com/o3de/o3de/commit/ea1c05cd8a0e4916881bdf9cbbc66d0bc7188499)

What this means for you, is that the nullptr checks aren't needed and you can skip the if check
```suggestion
        m_applicationImplFactory = CreateApplicationImplementationFactory();
        AZ::Interface<Application::ImplementationFactory>::Register(m_applicationImplFactory);
```